diff --git a/wand1.c b/wand1.c
index aa0f874..58eb084 100755
--- a/wand1.c
+++ b/wand1.c
@@ -3,6 +3,7 @@
 #include	<string.h>
 #include	<time.h>
 #include	<unistd.h>
+#include	<signal.h>
 #include	"wanddef.h"
 /*
 **      WANDER -- Non-deterministic fantasy story tool
@@ -29,6 +30,8 @@ uid_t	myeuid() { return(geteuid()); }	// normally in glib
 
 extern	long    wtell();
 extern	FILE	*fopen(), *wopen();
+extern	int		wseek(FILE *fp, long addr, int mode);
+extern	int		wclose(FILE *fp);
 
 int
 main(argc, argv)
diff --git a/wand2.c b/wand2.c
index c403e01..3a6c925 100755
--- a/wand2.c
+++ b/wand2.c
@@ -1,6 +1,9 @@
 #include	<stdio.h>
 #include	<stdlib.h>
 #include	<string.h>
+#include 	<fcntl.h>
+#include 	<time.h>
+#include 	<unistd.h>
 #include	"wanddef.h"
 /*
 **      WANDER -- Non-deterministic fantasy story tool
@@ -24,6 +27,8 @@ extern	long    ungotaddr;        /* addr of begin of last ungetlin() line */
 extern	char	*ctime();
 extern	long    wtell();
 extern	FILE	*fopen(), *wopen();
+extern	int		wseek(FILE *fp, long addr, int mode);
+extern	int		wgetc(FILE *fp);
 
 void
 restart(name)                       /* (init) restart from specified files */
@@ -77,30 +82,31 @@ takeobj(obj)
 }
 
 char	*
-objdesc(pre, art, wp, post)  /* assemble an object description in buf */
-char	*pre, *art, *post;
+objdesc(pre, art, wp, post, buf, bufsize)  /* assemble an object description in buf */
+char	*pre, *art, *post, *buf;
+int	bufsize;
 struct  wrdstr  *wp;
 {
-	char *cp, buf[1024];
+	char *cp;
 
 	if (wp->w_flg & W_ASIS) {		// don't augment the description
 	    cp = wp->w_word;
 	} else {				// do augment the description
 	    if (pre == (char *) 0) {           /* implies location description */
-		cp = cpyn(buf, thereis[class(wp)], sizeof(buf) - 1);
+		cp = cpyn(buf, thereis[class(wp)], bufsize - 1);
 	    } else {                             /* not a location description */
-		cp = strncpy(buf, pre, sizeof(buf) - 1);
+		cp = strncpy(buf, pre, bufsize - 1);
 		if (wp->w_flg & W_DONLY)
 		    wp++;
 	    }
 	    if ((wp->w_flg & (W_NOART | W_ASIS)) == 0) {  /* if article needed */
 		if (art)				      /* specified article */
-		    cp = cpyn(cp, art, buf + sizeof(buf) - cp - 1);
+		    cp = cpyn(cp, art, buf + bufsize - cp - 1);
 		else					/* default article */
-		    cp = cpyn(cp, aansome[class(wp)], buf + sizeof(buf) - cp - 1);
+		    cp = cpyn(cp, aansome[class(wp)], buf + bufsize - cp - 1);
 	    }
-	    cp = cpyn(cp, wp->w_word, buf + sizeof(buf) - cp - 1);
-	    cpyn(cp, post, buf + sizeof(buf) - cp - 1);
+	    cp = cpyn(cp, wp->w_word, buf + bufsize - cp - 1);
+	    cpyn(cp, post, buf + bufsize - cp - 1);
 	    cp = buf;
 	}
 	return(deparity(cp));
@@ -278,7 +284,7 @@ store(string)		// copy string into storebuf and return a pointer to it
 char	*string;
 {
         char *cp;
-	extern char *sbrk();
+	// extern char *sbrk();
 
 	if (storebp == 0) {
 	    if ((storebp = storebuf = sbrk(storenleft = 4096)) == (char *) -1) {
@@ -436,7 +442,7 @@ restore(char *file, int flag)
 {
 	char *cp, newname[PATHLENGTH];
 	int then, diff, i, fh;
-	extern char *sbrk();
+	// extern char *sbrk();
 
 	if (*file == '\0') {
 	    for (cp = file = curname; *cp; )
diff --git a/wanddef.h b/wanddef.h
index fad3961..d03fd18 100755
--- a/wanddef.h
+++ b/wanddef.h
@@ -12,8 +12,8 @@
 **          (also check `wandglb.c' for other modifiable parameters).
 */
 
-#define GAMESPATH(X)	"/Users/psl/Games+Toys/X"
-#define	WANDPATH(X)	GAMESPATH(Wander/X)			  /* (MOD) */
+#define GAMESPATH(X)	"wander_src/"#X
+#define	WANDPATH(X)	GAMESPATH(X)			  /* (MOD) */
 
 		      /* numbers in [] are #bytes data space used for each */
 #define MAXLOCS     1024                        /* [2] max locations (MOD) */
@@ -137,7 +137,8 @@ struct  paramstr {
 	int     p_time;		// when this file was saved
 	off_t	p_msize;	// size of the .misc file
 	off_t	p_wsize;	// size of the .wrld file
-} param;
+};
+extern	struct	paramstr    param;		    /* defined in wandglb.c */
 
 struct  placestr {
 	int     p_loc;
